client/tests/simulator/mod.rs

```diff
@@ -0,0 +1,9 @@
+#![cfg(feature = "tpm-simulator-tests")]
+
+// Please see "simulator_tests.rs" for details.
```

Comment:
Yeah, this is where we get into something I dislike about current rust language - the `tests` directory is kind of magical, in that it does something that we wouldn't expect with module resolution...

What I wish we could do is:

*   client/tests/simulator_tests/main.rs
*   client/tests/simulator_tests/submodule_a.rs
*   client/tests/simulator_tests/submodule_b.rs

But you can't do that. Instead you have to create a helper module and import that (like you're doing here). This is fine, and allows us to structure and organize our code, but the naming feels a little odd. Naming it `simulator` sounds like it's a module for providing the simulator, not for providing tests relating to the simulator.

---

So there's two options here:

1.  Turn `simulator` into a shared module that provides code for how to interact with the simulator, and then each test is it's own separate integration test (that's a bit annoying, I think; though we can reconsider if we want this in a future PR, we probably shouldn't change to this for now).
2.  Name the module explicitly to suggest it's about organizing test code for `simulator_tests`.

This might be bikeshedding a bit, so feel free to ignore, but WDYT of naming this module `simulator_tests_registration`, to make it clear it's a helper module that is responsible for organizing and registering tests related to `simulator_tests`.

So the final layout would look like this:

*   client/tests/simulator_tests_registration/mod.rs
*   client/tests/simulator_tests_registration/random.rs
*   client/tests/simulator_tests_registration/capability.rs
*   client/tests/simulator_tests.rs

Then in `simulator_tests.rs` we include that module:

```rs
mod simulator_tests_registration;
```

---

This is basically exactly what you're doing here, except that I'm suggesting we rename the `simulator` module to `simulator_tests_registration` to imply a more specific relation to `simulator_tests`.

Reply:
Super suggestion - I will adopt 🙂 I don't mind the "bikeshedding" (a word I had to look up, due to English being my 2nd language 😄 ).  It is important to get these details in place from the start, to avoid larger cleanups in the future 👍